### DISCUSSION TITLE
How can I compile `.vue` file just like `.ts` file using vue-tsc

### DISCUSSION THREAD

**Discussion by jd-solanki** [vuejs/language-tools#809] [Q&A] [answered]
Hi,

Recently I found that I can compile react project written in TS using `tsc` and get the equivalent JS code of my source code in `dist` dir.

I wanted to know can I do the same with the `vue-tsc` to get the JS compiled version of my vue TS project including vue SFC?

Regards.

**Comment by jd-solanki** [comment]
When I run `yarn vue-tsc --outDir dist-js -p tsconfig.json --noEmit` it throws below error:
![Screenshot 2021-12-21 at 7 38 03 PM](https://user-images.githubusercontent.com/47495003/146943323-9fe198a1-3c5a-47b8-81d1-c6345746493c.png)

I already added node_modules in `exclude` option in `tsconfig.json` file.

**Comment by ghost** [comment, marked as answer]
Is this what you're looking for? https://github.com/johnsoncodehk/volar/discussions/640

**Reply by jd-solanki** [reply]
I already checked that but I don't know which command should I run to achieve this?

Can you give me some suggestions on this?

Thanks.

**Reply by jd-solanki** [reply]
BTW, I am using vite

**Reply by johnsoncodehk** [reply]
This is not part of vue-tsc's responsibility. Type-check and emit js require different TS codes. Using vue-tsc will not generate correct component js logic code.
